### DISCUSSION TITLE
SnipeIT: Update config from IP to Domain and set up SSL with or without reverse proxy.

### DISCUSSION THREAD

**Discussion by michelroegl-brunner** [community-scripts/ProxmoxVE#671] [Guides]
To change how SnipeIT is reachable you need to change two files.

The first one you need to alter is the nginx config file.
`vim /etc/nginx/conf.d/snipeit.conf`

Here you need to change the value **server_name**. Change _snipeit.example.com_ to your domain name.
`server_name snipeit.example.com`

The second file you need to change is the SnipeIT Webapp config.
`vim /opt/snipe-it/.env `

Here you need to change the value **APP_URL". _Change http://snipeit.exampel.com_ to your domain name.
`APP_URL=http://snipeit.example.com`

This are alle the changes needed for HTTP.

For HTTPS with reverse Proxy you need to set `proxy_set_header X-Forwarded-Proto $scheme;` (NGINX) or `RequestHeader set X-Forwarded-Proto "https"` (APACHE) at you proxy config. Below as an exmple are the settings for  [Nginx Proxy Manager](https://community-scripts.github.io/ProxmoxVE/scripts?id=nginxproxymanager)

![nginx_reverse_proxy_snipeit2](https://github.com/user-attachments/assets/88573848-a1fd-41af-8d36-fa98039f9bbc)
![nginx_reverse_proxy_snipeit](https://github.com/user-attachments/assets/928f593b-31f4-47cb-8af3-cb918ae5dbc2)

And in the SnipeIT Webapp config you need to set **APP_TRUSTED_PROXIES** to the IP-Address of your proxy.
`vim nano /opt/snipe-it/.env `

To get HTTPS working without reverse proxy you need to change the nginx config file.
And again change the **server_name** from _snipeit.exampe.com_ to your own domain.

`vim /etc/nginx/conf.d/snipeit.conf` 

Replace all content with:

```
server {
    listen 80;
    server_name snipeit.example.com;
    return 301 https://$server_name$request_uri;
}
server {
    listen 443 ssl;
    server_name snipeit.exampe.com;
    ssl_certificate /path/to/your.crt;
    ssl_certificate_key /path/to/your.key;
    ssl_protocols SSLv3 TLSv1 TLSv1.1 TLSv1.2;
    ssl_prefer_server_ciphers on;
    ssl_ciphers ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-RC4-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:RC4-SHA:!aNULL:!eNULL:!EXPORT:!DES:!3DES:!MD5:!DSS:!PKS;
    ssl_session_timeout 5m;
    ssl_session_cache builtin:1000 shared:SSL:10m;
    root /opt/snipe-it/public/;
    index index.php index.html index.htm;
    location / {
        try_files $uri $uri/ /index.php$is_args$args;
    }
    location ~ \.php$ {
        try_files $uri $uri/ =404;
        fastcgi_pass unix:/var/run/php8.2-fpm-www.sock;
        fastcgi_index index.php;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include fastcgi_params;
    }
}
```

**Comment by ismaelster** [comment]
Hello Michel, thank you for the post-installation guide. I have followed it completely but I have a problem when using Nginx Proxy Manager:

I have correctly created the A record in my domain manager.
In the .env file: app_url I have the complete application address: http://192.168.x.x and in the snipeit.conf file: server_name I also have the complete application address: http://192.168.x.x
I have added the IP of my Nginx Proxy Manager in the .env file: app_trusted_proxies: 192.168.x.x
I have added in my Nginx Proxy Manager, as you indicate, the line proxy_set_header X-Forwarded-Proto $forward_scheme;
But unfortunately, I can access the application locally via the IP 192.168.x.x but not through the domain snipeit.domain.com.

What am I missing?

**Comment by ismaelster** [comment]
I forgot to attach the message I receive when I try to connect via https

<img width="577" alt="Captura de pantalla 2024-12-18 a las 19 10 27" src="https://github.com/user-attachments/assets/cbf7bb24-3dfc-4408-b684-191c77c4b470" />

**Comment by dajmlj812** [comment]
I am getting the same. 
![2024-12-22_09-23-43](https://github.com/user-attachments/assets/333ec6b3-8860-43ee-a3a8-d436fdd6e736)
![2024-12-22_09-24-39](https://github.com/user-attachments/assets/baf196cc-435a-49a6-90f6-52ba1030cd97)
![2024-12-22_09-26-26](https://github.com/user-attachments/assets/8888394f-28c5-4e50-972c-4a11088c2c11)
![2024-12-22_09-27-16](https://github.com/user-attachments/assets/1468866f-aded-427e-998d-2fe1cbb3e4b4)
![2024-12-22_09-27-51](https://github.com/user-attachments/assets/f782ee28-f8f3-4473-83a0-77ae6ea4e528)
![2024-12-22_09-29-24](https://github.com/user-attachments/assets/696249c7-ac03-456f-a7f4-51c94d9cac73)

**Comment by dajmlj812** [comment]
Nevermind. Fixed it by updating to https://...
![2024-12-22_09-33-08](https://github.com/user-attachments/assets/163f0f8f-2838-4506-81ab-77baeeb82439)
![2024-12-22_09-35-18](https://github.com/user-attachments/assets/74816f7f-a176-425f-83de-65c8b094b323)

**Reply by ismaelster** [reply]
Hi, I see that is it working for you, I tried several things before without no lucky, but today tried a different thing, I made a dnslookup with the domain and I see 2 ips, one for the nginx proxy manager thas was setup in .env and another ip from the web server (hosting), I have write this ip in the .env and works!

so, I dont know if this tip could be helpfully for another people :)

**Comment by sanoj-m** [comment]
The Latest SnipeIT update v8.3.5 looks to be braking all the asset images, i guess there are some configuration changes while using the 'update' feature.

**Reply by michelroegl-brunner** [reply]
In the changelog i am not able to find any changes to Asset Images or Config changes. Can you point out where you find that?

**Comment by sanoj-m** [comment]
https://github.com/user-attachments/assets/d030c59c-e5d1-4a61-b44a-ec0552139fdf

There are couple of duplicated folders, maybe its from snipeit itself, this new path changes the locations for assets and attachments.

also, i had done some changes to the ngnix and it got updated with default values which i had to comment off to get the service starting.

<img width="746" height="434" alt="image" src="https://github.com/user-attachments/assets/e0fcd9de-462d-4820-8a8e-939360228db1" />

**Comment by icf20** [comment]
i run the update and this not longer works, whats the deal with this? config incompatible?

**Reply by ruanmed** [reply]
What was your previous version before update?
I just ran an upgrade from version 8.3.5 to 8.3.7 successfully (using the update command in the LXC). (only side effect seems that the theme automatically changed from the normal blue/light I was used to a green/dark theme, I'll check how to change that). 

There was a breaking change recently in SnipeIT that changed the assets folders, from 8.2.x to 8.3.x, like other users described above.
However, my installation did not break, I just had to move some of the private_uploads folders iirc because images were not working.

Do you have logs from the upgrade you performed? And you can no longer access/login to your snipe-it instance?

Edit: It seems like during the upgrade files from both `public/uploads` and `storage/private_uploads` go missing. Looking at the update script for snipeit I think the copy commands are creating a nested directory.. so that's probably why it's happening, I think I'll submit a possible fix PR.

**Comment by racortesl** [comment]
I have an existing Snipe-IT LXC running locally without any problem, but now I need access from outside and I'm using Nginx Proxy Manager. I configured my .env and snipeit.conf files and get access from the public interface. I can login into my accounts, see the list of all the assets, models, vendors, etc. BUT if I open any user, asset or anything, it shows me a 404 error page. Also, If I tried to create a new asset, it won't list the vendors, models or anything, and if I generate a new, I get a 404 error page when I tried to open it.

It looks like the db were corrupted, but if I roll back to the local ip access, I can use the app without any problem.

<img width="573" height="202" alt="Screenshot 2026-01-28 084703" src="https://github.com/user-attachments/assets/b0dabc68-2b30-4876-afa9-290fbef458ec" />

<img width="606" height="260" alt="Screenshot 2026-01-28 084834" src="https://github.com/user-attachments/assets/f43e078e-ac1d-4e50-8cb3-b5c716c7cd58" />

<img width="1183" height="670" alt="Screenshot 2026-01-28 085041" src="https://github.com/user-attachments/assets/a9269107-965d-494f-8aa2-fe0a429a7d07" />

<img width="960" height="570" alt="Screenshot 2026-01-28 085716" src="https://github.com/user-attachments/assets/5a3a685a-8f49-4e47-9b15-895010aaa05c" />

**Reply by michelroegl-brunner** [reply]
That is a issue for the SnipeIT Repo and not here.

**Reply by racortesl** [reply]
Thanks. I'll post the issue there.

**Comment by jonasfessler** [comment]
Hi, 
so I had the same issue with Nginx only displaying the default "further configuration is requred" message. 
After trying out different approaches to this I finally fixed it by just straight up replacing the default configuration in both 
`/etc/nginx/sites-enabled`
and
`/etc/nginx/sites-available`
with the `snipeit.conf` from `/etc/nginx/conf.d`
and then renaming it to default to overwrite nginx default configuration :)

While that is only a temporary fix (we call it Pfusch in german) it definitly works.

For a quick-and-dirty copy-paste:

```
cp /etc/nginx/conf.d/snipeit.conf /etc/nginx/sites-available
cp /etc/nginx/conf.d/snipeit.conf /etc/nginx/sites-enabled
rm /etc/nginx/sites-available/default
rm /etc/nginx/sites-enabled/default
mv /etc/nginx/sites-available/snipeit.conf /etc/nginx/sites-available/default
mv /etc/nginx/sites-enabled/snipeit.conf /etc/nginx/sites-enabled/default
```

While this makes SnipeIT available by accessing the IP in your browser I do not recommend doing that. 
Also this may or may not only work with HTTP (didn't test)
